core/lib/zksync_core/src/eth_sender/eth_tx_aggregator.rs

```diff
@@ -496,15 +513,20 @@ impl EthTxAggregator {
     async fn get_next_nonce(
         &self,
         storage: &mut StorageProcessor<'_>,
+        from_addr: Option<Address>,
     ) -> Result<u64, ETHSenderError> {
         let db_nonce = storage
             .eth_sender_dal()
-            .get_next_nonce()
+            .get_next_nonce(from_addr)
             .await
             .unwrap()
             .unwrap_or(0);
         // Between server starts we can execute some txs using operator account or remove some txs from the database
         // At the start we have to consider this fact and get the max nonce.
-        Ok(db_nonce.max(self.base_nonce))
+        Ok(if from_addr.is_none() {
+            db_nonce.max(self.base_nonce)
+        } else {
+            db_nonce
```

Comment:
> Between server starts we can execute some txs using operator account or remove some txs from the database

Can this logic be applied to the "overwritten" address? That is, is it the right thing to just take the nonce from DB? How do we set base_nonce - wouldn't it be possible to do the same for `from_addr`?

Reply:
good point, fixed.